dbuf-core/src/parser/lexer.rs

```diff
@@ -1,8 +1,10 @@
-use chumsky::span::SimpleSpan;
-use logos::Logos;
+use std::{cell::RefCell, rc::Rc};
+
+use logos::{Lexer, Logos, Skip};
 use unescape::unescape;
 
 #[derive(Logos, Debug, PartialEq, Clone)]
+#[logos(extras = Rc<RefCell<Vec<usize>>>)]
```

Comment:
It should be simpler than that, storing line lengths is very extraneous

I think instead of using logos' `extras`, we need to replicate what is done in their `.spanned` helper implementation, but compute Offsets instead of just span lengths